### PR TITLE
Add `fname` to subscriptions in memory

### PR DESCRIPTION
**Pull request by rodrigok**
@RocketChat/core 

- [x] Subscriptions
- [x] Messages
- [x] Remove undesired updates
- [x] Update name of messages when user change his name
- [x] Only do this if the Real Name setting is enabled
- [x] Do not remove name from subscriptions on update

### CODE REVIEW

File: server/methods/loadHistory.js [hunk 1]
```diff
@@ -58,10 +58,16 @@ Meteor.methods({
...
 		}
 
+		const UI_Use_Real_Name = RocketChat.settings.get('UI_Use_Real_Name') === true;
+
 		const messages = records.map((message) => {
 			message.starred = _.findWhere(message.starred, {
 				_id: fromId
 			});
+			if (message.u && message.u._id && UI_Use_Real_Name) {
+				const user = RocketChat.models.Users.findOneById(message.u._id);
```
Comment (alexbrazier) [inline server/methods/loadHistory.js]: Isn't this going to make quite a lot of request to the db to get the users name?
Would it be better to collect an array of user ids then fetch the full names all at once outside the loop?

File: server/stream/messages.js [hunk 2]
```diff
@@ -45,6 +45,10 @@ msgStream.allowEmit('__my_messages__', function(eventName, msg, options) {
 Meteor.startup(function() {
 	function publishMessage(type, record) {
 		if (record._hidden !== true && (record.imported == null)) {
+			if (record.u && record.u._id) {
```
Comment (alexbrazier) [inline server/stream/messages.js]: Should this check for the `UI_Use_Real_Name` setting before doing this?

File: packages/rocketchat-lib/server/methods/getChannelHistory.js [hunk 3]
```diff
@@ -49,8 +49,14 @@ Meteor.methods({
 			records = RocketChat.models.Messages.findVisibleByRoomIdBetweenTimestamps(rid, oldest, latest, options).fetch();
 		}
 
+		const UI_Use_Real_Name = RocketChat.settings.get('UI_Use_Real_Name') === true;
+
 		const messages = _.map(records, (message) => {
 			message.starred = _.findWhere(message.starred, { _id: fromUserId });
+			if (message.u && message.u._id && UI_Use_Real_Name) {
+				const user = RocketChat.models.Users.findOneById(message.u._id);
```
Comment (alexbrazier) [inline packages/rocketchat-lib/server/methods/getChannelHistory.js]: Same as my other comment, would it be better to fetch all users in result at once, then map the name on to `message.u.name`?

File: server/methods/loadHistory.js (hunk 1 above)
Comment (rodrigok) [inline server/methods/loadHistory.js]: Not to the DB, to the memory cache, it's really fast cuz is indexed using a hash map

File: packages/rocketchat-lib/server/methods/getChannelHistory.js (hunk 3 above)
Comment (rodrigok) [inline packages/rocketchat-lib/server/methods/getChannelHistory.js]: Same answer 😄

**Comment by amenk** [comment]
Hm, where is the real name setting (Running 0.55.1)

**Comment by amenk** [comment]
But the user ist is sorted by user name, not real name? That is a bit strange ...

**Comment by amenk** [comment]
Also the quick search (Ctrl+K) does not seem to take the fullname into account

**Comment by alexbrazier** [comment]
@amenk The order has been resolved in #6758 
As for the quick search, it seems to be working for me, but only in the server results, so a fix might be needed to make it faster on the frontend. Does it not work at all for you?

**Pull request merged**